classes/local/users.php

```diff
@@ -66,19 +66,23 @@ public function get_users_with_credentials($enrolledusers, $groupid = null) {
 
         $certificatesmemo = array();
         if ($groupid) {
-            $credentialsclient = new credentials($this->apirest);
-            $certificates = $credentialsclient->get_credentials($groupid);
+            try {
+                $credentialsclient = new credentials($this->apirest);
+                $certificates = $credentialsclient->get_credentials($groupid);
 
-            foreach ($certificates as $certificate) {
-                if (isset($certificate->url)) {
-                    $credentialurl = $certificate->url;
-                } else {
-                    $credentialurl = 'https://www.credential.net/' . $certificate->id;
+                foreach ($certificates as $certificate) {
+                    if (isset($certificate->url)) {
+                        $credentialurl = $certificate->url;
+                    } else {
+                        $credentialurl = 'https://www.credential.net/' . $certificate->id;
+                    }
+                    $certificatesmemo[$certificate->recipient->email] = array(
+                        'credentialid' => $certificate->id,
+                        'credentialurl' => $credentialurl
+                    );
```

Comment:
I'd move `foreach ($certificates as $certificate) { ... }` to outside of `try { ... } catch { ... } ` because I think we  want to catch errors only from `$credentialsclient->get_credentials($groupid)`.